tools/dependency/release_dates.py

```diff
@@ -40,32 +42,58 @@ def verify_and_print_latest_release(dep, repo, metadata_version, release_date):
         latest_release = repo.get_latest_release()
         if latest_release.created_at > release_date and latest_release.tag_name != metadata_version:
             print(
-                f'*WARNING* {dep} has a newer release than {metadata_version}@<{release_date}>: '
-                f'{latest_release.tag_name}@<{latest_release.created_at}>')
-    except github.UnknownObjectException:
+                Fore.YELLOW
+                + f'*WARNING* {dep} has a newer release than {metadata_version}@<{release_date}>: '
+                f'{latest_release.tag_name}@<{latest_release.created_at}>' + Style.RESET_ALL)
+    except github.GithubException as e:
         pass
 
 
 # Print GitHub release date, throw ReleaseDateError on mismatch with metadata release date.
 def verify_and_print_release_date(dep, github_release_date, metadata_release_date):
     mismatch = ''
     iso_release_date = format_utc_date(github_release_date)
-    print(f'{dep} has a GitHub release date {iso_release_date}')
+    print(Fore.GREEN + f'{dep} has a GitHub release date {iso_release_date}' + Style.RESET_ALL)
     if iso_release_date != metadata_release_date:
         raise ReleaseDateError(f'Mismatch with metadata release date of {metadata_release_date}')
 
 
 # Extract release date from GitHub API.
 def get_release_date(repo, metadata_version, github_release):
     if github_release.tagged:
-        tags = repo.get_tags()
-        for tag in tags:
-            if tag.name == github_release.version:
-                return tag.commit.commit.committer.date
+
+        try:
+            latest = repo.get_latest_release()
+        except github.GithubException as e:
+            latest = ''
+            pass
+
+        if latest and (github_release.version <= latest.tag_name):
+            release = repo.get_release(github_release.version)
+            return release.published_at
+        else:
+            tags = repo.get_tags()
+            current_metadata_tag_commit_date = ''
+            for tag in tags.reversed:
+                if tag.name == github_release.version:
+                    current_metadata_tag_commit_date = tag.commit.commit.committer.date
+                if not (version.parse(tag.name).is_prerelease) and version.parse(
+                        tag.name) > version.parse(github_release.version):
```

Comment:
nit: i reallly dislike multiline `if` clauses - my suggestion is (always) to express the logic in a variable - eg:

```
has_newer_release_date = (
    not version.parse(tag.name).is_prerelease
    and version.parse(tag.name) > version.parse(github_release.version))

if has_newer_release_date:
   ...do something...

```